filestore/dataobj.go

```diff
@@ -0,0 +1,136 @@
+package filestore
+
+import (
+	"fmt"
+	pb "github.com/ipfs/go-ipfs/filestore/pb"
+	"math"
+	"time"
+)
+
+const (
+	// If NoBlockData is true the Data is missing the Block data
+	// as that is provided by the underlying file
+	NoBlockData = 1
+	// If WholeFile is true the Data object represents a complete
+	// file and Size is the size of the file
+	WholeFile = 2
+	// If the node represents an a file but is not a leaf
+	// If WholeFile is also true than it is the file's root node
+	Internal = 4
+	// If the block was determined to no longer be valid
+	Invalid = 8
+)
+
+type DataObj struct {
+	Flags uint64
+	// The path to the file that holds the data for the object, an
+	// empty string if there is no underlying file
+	FilePath string
+	Offset   uint64
+	Size     uint64
+	ModTime  float64
+	Data     []byte
```

Comment:
```
Execute bool
Write   bool
```

should be added here.

Explanation:

- **execute bit** whether the execute bit will be set for the OS (fuse)
- **write bit** whether it is possible to alter the content of the file (thus recreating a new version of it) through fuse, it can be set using e.g. `chmod`